### ISSUE TITLE
Disconnections because of empty data received when under heavy txps load

### ISSUE THREAD

**Issue by litchipi** [massalabs/massa#4296]
**Describe the bug**
When under a load of 3000 txps or more, a node gets random disconnections from the remote note because of empty data received.

**To Reproduce**
Start a labnet on `testnet_25` with txps of 3000 or more

Watch how it disconnects from time to time from the labnet node.

The heavier the load is, the more frequent it happens apparently

**Expected behavior**
Should stay stable

**Additional context**
I like trains

**Comment by damip** [comment]
@litchipi does it still happen when the stream limiter is disabled ?

**Comment by litchipi** [comment]
@damip Yes it does... When running `testnet_25` with a version of PeerNet without the stream limiter, the exact same behavior can be observed :disappointed:
(branches `debug_highload_disconnect` on massa, `debug_highload` on Peernet)

**Comment by damip** [comment]
what causes the disconnection precisely ? read() == 0 ?

**Comment by litchipi** [comment]
It's on the reader loop of the peer in PeerNet, (`peer.rs` line ~320)
It either raise an error `"Os { code: 11, kind: WouldBlock, message: \"Resource temporarily unavailable\" }"` or `Os { code: 104, kind: ConnectionReset, message: \"Connection reset by peer\" }`
(I also had an error related to empty read, but didn't have it on last test apparently)

**Comment by damip** [comment]
`kind: WouldBlock` => are we sure that we set `set_nonblocking(false)` both on incoming and outgoing connections ?

**Comment by damip** [comment]
when getting a WOULDBLOCK we should:
* print whether the socket is incoming or outgoing
* print the nonblocking state of the socket and make sure that it is FALSE

**Comment by litchipi** [comment]
`set_nonblocking(false)` is used on every stream created in PeerNet (in the `set_tcp_stream_config` function)
I'll make some tests after adding the debug prints where the error occurs

**Comment by litchipi** [comment]
```
[/home/john/workspace/PeerNet/src/peer.rs:321] "TIM     receive error:" = "TIM     receive error:"
[/home/john/workspace/PeerNet/src/peer.rs:321] &e = PeerNetErrorData {
    location: "error read data stream",
    error_type: ReceiveError,
    error: None,
    add_msg: Some(
        "Os { code: 11, kind: WouldBlock, message: \"Resource temporarily unavailable\" }",
    ),
}
[/home/john/workspace/PeerNet/src/transports/endpoint.rs:23] config = TcpConnectionConfig {
    rate_limit: 18446744073709551615,
    rate_time_window: 1s,
    rate_bucket_size: 18446744073709551615,
    data_channel_size: 10000,
    max_message_size: 1048576000,
    write_timeout: 5s,
    read_timeout: 5s,
}
[/home/john/workspace/PeerNet/src/transports/endpoint.rs:24] address = 145.239.66.206:31244
[/home/john/workspace/PeerNet/src/transports/endpoint.rs:25] stream_limiter = TcpStream {
    addr: 192.168.1.220:47408,
    peer: 145.239.66.206:31244,
    fd: 45,
}
Removing connection from: PeerId { public_key: P1kKfgrCveVnosUkxTzaBw5cf9f2cbTvK3R5Ssb2Pf76au8xwmH }
```

**Comment by litchipi** [comment]
The problem is local, we cannot reproduce the issue with another computer.
I have several ideas what's happening, i'll debug this locally
